src/frontend/src/icp-eth/components/tokens/IcManageTokens.svelte

```diff
@@ -69,32 +68,28 @@
 			);
 
 	let tokens: IcrcCustomToken[] = [];
-	$: tokens = filteredTokens.map(({ ledgerCanisterId, enabled, ...rest }) => ({
-		ledgerCanisterId,
-		enabled: modifiedTokens[`${ledgerCanisterId}`]?.enabled ?? enabled,
+	$: tokens = filteredTokens.map(({ id, network, enabled, ...rest }) => ({
+		id,
+		network,
+		enabled: modifiedTokens[`${network.id.description}-${id.description}`]?.enabled ?? enabled,
 		...rest
 	}));
 
 	let noTokensMatch = false;
 	$: noTokensMatch = tokens.length === 0;
 
-	let modifiedTokens: Record<CanisterIdText, IcrcCustomToken> = {};
-	const onToggle = ({
-		detail: { ledgerCanisterId, enabled, ...rest }
-	}: CustomEvent<IcrcCustomToken>) => {
-		const { [`${ledgerCanisterId}`]: current, ...tokens } = modifiedTokens;
+	let modifiedTokens: Record<string, IcrcCustomToken> = {};
+	const onToggle = ({ detail: { id, network, ...rest } }: CustomEvent<IcrcCustomToken>) => {
+		const { id: networkId } = network;
+		const { [`${networkId.description}-${id.description}`]: current, ...tokens } = modifiedTokens;
```

Comment:
Cool idea. Not sure, maybe it's too much and not useful but, we can also create a symbol. But maybe that's too much and not useful. Just sharing in case the question I asked my self would not be too dumb 😅.

```
const { [Symbol(`${networkId.description}-${id.description}`)]: current, ...tokens } = modifiedTokens;

Reply:
the only particularity with symbols is that apparently if you use `Object.keys` it returns an empty list, you should use `Object.getOwnPropertySymbols` instead

I found out yesterday searching, I cannot find the link anymore... but I tested and it really returned empty ([remember this comment?](https://github.com/dfinity/oisy-wallet/pull/1477#issuecomment-2176327858))


I really have no preference, it was just to reduce the code that changed, you tell me if you feel strong on that, @peterpeterparker

Reply:
Agree with you, that was just an idea/question that crossed my mind.